### PR TITLE
adding postgresqlconfiguration facts

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY
Adding PostgreSQL configuration facts module

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME

azure_rm_postgresqlconfiguration_facts

##### ANSIBLE VERSION
2.7

### CODE REVIEW

**Comment by Fred-sun** [comment]
kindly ping

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlconfiguration_facts.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+module: azure_rm_postgresqlconfiguration_facts
+version_added: "2.8"
+short_description: Get Azure PostgreSQL Configuration facts.
+description:
+    - Get facts of Azure PostgreSQL Configuration.
+
+options:
+    resource_group:
+        description:
+            - The name of the resource group that contains the resource. You can obtain this value from the Azure Resource Manager API or the portal.
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_postgresqlconfiguration_facts.py]: the second sentence seems useless.

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlconfiguration_facts.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    description: A list of dictionaries containing MySQL Server settings.
+    returned: always
+    type: complex
+    contains:
+        id:
+            description:
+                - Setting resource ID
+            returned: always
+            type: str
+            sample: "/subscriptions/ffffffff-ffff-ffff-ffff-ffffffffffff/resourceGroups/testrg/providers/Microsoft.DBforPostgreSQL/servers/testpostgresqlser
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_postgresqlconfiguration_facts.py]: xxxx-xxxx ... for consistent?

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlconfiguration_facts.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        self.mgmt_client = None
+        self.resource_group = None
+        self.server_name = None
+        self.setting_name = None
+        super(AzureRMPostgreSQLConfigurationFacts, self).__init__(self.module_arg_spec, supports_tags=False)
+
+    def exec_module(self, **kwargs):
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+        self.mgmt_client = self.get_mgmt_svc_client(PostgreSQLManagementClient,
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_postgresqlconfiguration_facts.py:138]: no api version either?

File: test/integration/targets/azure_rm_postgresqlserver/tasks/main.yml [hunk 4]
```diff
@@ -41,24 +41,24 @@
...
-    enforce_ssl: True
-    admin_username: zimxyz
-    admin_password: Testpasswordxyz12!
-  register: output
-- name: Assert the state has not changed
-  assert:
-    that:
-      - output.changed == false
-      - output.state == 'Ready'
+#- name: Create again instance of PostgreSQL Server
```
Comment (yungezz) [inline test/integration/targets/azure_rm_postgresqlserver/tasks/main.yml]: why disalbe? what's the 500 error mean? could you pls confirm there's no problem in postgresqlserver module?

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlconfiguration_facts.py (hunk 3 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_postgresqlconfiguration_facts.py:138]: yes the same

File: test/integration/targets/azure_rm_postgresqlserver/tasks/main.yml (hunk 4 above)
Comment (zikalino) [reply test/integration/targets/azure_rm_postgresqlserver/tasks/main.yml]: error 500 means internal error of the service.
this was working before it just stopped working now.
so the module is correct, there's a service problem.
the test was entirely disabled, now I reenabled it but disabled only failing parts.

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlconfiguration_facts.py (hunk 2 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_postgresqlconfiguration_facts.py]: yes, i think we should agree on common "placeholder" value, now we have a few different....

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlconfiguration_facts.py [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+                required=True
+            ),
+            name=dict(
+                type='str'
+            )
+        )
+        # store the results of the module operation
+        self.results = dict(
+            changed=False,
+            ansible_facts=dict()
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_postgresqlconfiguration_facts.py]: so ansible_facts back? i think Matt/Jordan insists to remove it

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlconfiguration_facts.py [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+        '''
+        response = None
+        results = []
+        try:
+            response = self.mgmt_client.configurations.get(resource_group_name=self.resource_group,
+                                                           server_name=self.server_name,
+                                                           configuration_name=self.name)
+            self.log("Response : {0}".format(response))
+        except CloudError as e:
+            self.log('Could not get facts for Configurations.')
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_postgresqlconfiguration_facts.py]: self.fail?

**Review by yungezz — APPROVED** [review]

**Comment by yungezz** [comment]
shipit

**Comment by yungezz** [comment]
shipit

**Pull request merged**